paasta_tools/cli/cmds/logs.py

```diff
@@ -503,18 +509,46 @@ def outer(log_reader_class):
     return outer
 
 
-def get_log_reader_class(name: str) -> Type["ScribeLogReader"]:
+def get_log_reader_class(
+    name: str,
+) -> Union[Type["ScribeLogReader"], Type["VectorLogsReader"]]:
     return _log_reader_classes[name]
 
 
 def list_log_readers() -> Iterable[str]:
     return _log_reader_classes.keys()
 
 
-def get_log_reader() -> "ScribeLogReader":
-    log_reader_config = load_system_paasta_config().get_log_reader()
-    log_reader_class = get_log_reader_class(log_reader_config["driver"])
-    return log_reader_class(**log_reader_config.get("options", {}))
+def get_log_reader(components: Set[str]) -> "LogReader":
+    log_readers_config = load_system_paasta_config().get_log_readers()
+    # ideally we should use a single "driver" for all components, but in cases where more than one is used for different components,
+    # we should use the first one that supports all requested components
+    # otherwise signal an error and suggest to provide a different list of components
+    components_lists = []
+    for log_reader_config in log_readers_config:
+        supported_components = log_reader_config.get("components", [])
+        components_lists.append(supported_components)
+        if components <= set(supported_components):
```

Comment:
imo, `issubset` is a little quicker to parse as a human (i had to look at the docs to remind myself what `<=` means for sets) - but i don't feel too strongly about this :)

Reply:
agree